### PR TITLE
Do not persist SourceMap contents in `FileInBundleCache`

### PR DESCRIPTION
**Pull request by Swatinem**
According to metrics, the cache is quite effective in ST and on LPQ instances. Less so in production, most likely because the cache is too small.

To increase the effective size of the cache, we do not store the actual sourcemap contents in it anymore. These contents are only used for sourcemapcache generation, and are not usually required.

#skip-changelog

### CODE REVIEW

File: crates/symbolicator-js/src/lookup.rs [hunk 1]
```diff
@@ -490,6 +484,7 @@ impl<T> CachedFileEntry<T> {
 /// the parts that we care about.
 #[derive(Clone)]
 pub struct CachedFile {
+    pub is_lazy: bool,
```
Comment (loewenheim) [inline crates/symbolicator-js/src/lookup.rs:487]: This should be printed in the `Debug` impl.

File: crates/symbolicator-js/src/lookup.rs [hunk 2]
```diff
@@ -966,10 +967,18 @@ impl ArtifactFetcher {
         }
 
         // First see if we have already cached this file for any of this event's bundles.
-        if let Some(file_entry) = self
+        if let Some((bundle_uri, file_entry, resolved_with)) = self
             .files_in_bundles
             .try_get(self.artifact_bundles.keys().rev().cloned(), key.clone())
         {
+            if let Some(Ok((_, bundle_resolved_with))) = self.artifact_bundles.get(&bundle_uri) {
```
Comment (anonrig) [inline crates/symbolicator-js/src/lookup.rs:974]: Can you add a comment?

File: crates/symbolicator-js/src/lookup.rs [hunk 3]
```diff
@@ -1292,46 +1307,16 @@ impl ArtifactFetcher {
...
-
-    fn load(&self, data: ByteView<'static>) -> CacheEntry<Self::Item> {
-        parse_sourcemap_cache_owned(data)
-    }
-}
-
-fn parse_sourcemap_cache_owned(byteview: ByteView<'static>) -> CacheEntry<OwnedSourceMapCache> {
-    SelfCell::try_new(byteview, |p| unsafe {
-        SourceMapCache::parse(&*p).map_err(CacheError::from_std_error)
+pub fn open_bundle(bundle: Arc<ObjectHandle>) -> CacheEntry<ArtifactBundle> {
```
Comment (anonrig) [inline crates/symbolicator-js/src/lookup.rs:1310]: Can you add a comment?

File: crates/symbolicator-js/src/sourcemap_cache.rs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+use crate::lookup::{
+    open_bundle, ArtifactBundle, ArtifactBundles, CachedFile, CachedFileUri, FileKey,
+    OwnedSourceMapCache,
+};
+use crate::utils::get_release_file_candidate_urls;
+
+#[derive(Clone, Debug)]
+pub enum SourceMapContents {
+    Lazy(Arc<ObjectHandle>, FileKey),
```
Comment (anonrig) [inline crates/symbolicator-js/src/sourcemap_cache.rs:23]: Can we select a better word for lazy?

**Review by loewenheim — APPROVED** [review]

**Pull request merged**